core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/MetricsServiceImpl.java

```diff
@@ -257,10 +262,16 @@ public void startUp(Session session, String keyspace, boolean resetDb, boolean c
                 .put(STRING, Functions::getStringDataPoint)
                 .build();
 
+        metricsIdxCache = CacheBuilder.newBuilder().expireAfterAccess(1, TimeUnit.HOURS).maximumSize(40000).build();
```

Comment:
We need to use Infinispan with the TTL changes also (and that was in a PR already as well). So it will be added sooner or later in any case, as we need to store the TTL value distributed in the cluster for deletion job. Might as well add it here so we don't have to rewrite this instantly.